engine/crates/engine-v2/schema/src/ids.rs

```diff
@@ -43,6 +45,14 @@ macro_rules! id_newtypes {
                 }
             }
 
+            impl std::ops::IndexMut<crate::ids::IdRange<$name>> for $ty {
+                fn index_mut(&mut self, range: crate::ids::IdRange<$name>) -> &mut Self::Output {
+                    let crate::ids::IdRange { start, end } = range;
+                    let start = usize::from(start);
+                    let end = usize::from(end);
+                    &mut self.$field[start..end]
+                }
+            }
```

Comment:
What's this being used for?  Wouldn't have thought we should be editing the schema generally....

Reply:
We might not in this PR, but I'm sorting the enum values by their name in a further PR. Same for field arguments. Currently the federated graph doesn't guarantee that, should likely be changed at some point.